### PR TITLE
feat: implement Ready status condition for ProjectDevelopmentStream

### PR DESCRIPTION
**Pull request by kelchen123**
Implement ready status condition to address KFLUXVNGD-472.

### CODE REVIEW

**Review by amisstea — COMMENTED** [review]
Looks good but it might be a good to introduce some unit tests to verify these status conditions are being set as expected.

**Comment by yftacherzog** [comment]
The EC failure should be fixed after a rebase.

File: internal/controller/projectdevelopmentstream_controller_test.go [hunk 1]
```diff
@@ -122,6 +135,154 @@ var _ = Describe("ProjectDevelopmentStream Controller", func() {
 			"projctl_v1beta1_pds_w_existing_comp.yaml",
 		),
 	)
+
+	Context("Status Conditions for edge cases", func() {
```
Comment (ifireball) [inline internal/controller/projectdevelopmentstream_controller_test.go]: I would expect to see the status condition tests as part of the existing YAML file-based tests we already have, why is that not the case?

File: internal/controller/projectdevelopmentstream_controller.go [hunk 2]
```diff
@@ -82,6 +88,11 @@ func (r *ProjectDevelopmentStreamReconciler) Reconcile(ctx context.Context, req
 	log = log.WithValues("PDS name", pds.ObjectMeta.Name)
 	log = muxr.NewMuxLogger(log, eventr.NewEventr(r.Recorder, &pds))
 
+	// Set initial condition
+	if err := r.setReadyCondition(ctx, &pds, metav1.ConditionUnknown, "Reconciling", "Reconciling ProjectDevelopmentStream"); err != nil {
+		log.Error(err, "Failed to update status")
```
Comment (ifireball) [inline internal/controller/projectdevelopmentstream_controller.go]: instead of adding multipel log calls here to log each time we fail to set the status condition, we can do the logging from the setReadyCondition function and keep the code here cleaner. We can pass the logger around by putting it into the context, there seem to be a convention of doing that.

File: internal/controller/projectdevelopmentstream_controller_test.go [hunk 3]
```diff
@@ -210,6 +363,23 @@ func dropUncomparableMetadata(obj *unstructured.Unstructured) {
...
+		return
+	}
+
+	for i := range conditions {
+		if condition, ok := conditions[i].(map[string]interface{}); ok {
+			delete(condition, "lastTransitionTime")
+		}
+	}
+
+	_ = unstructured.SetNestedSlice(obj.Object, conditions, "status", "conditions")
```
Comment (amisstea) [inline internal/controller/projectdevelopmentstream_controller_test.go]:
```suggestion
	Expect(unstructured.SetNestedSlice(obj.Object, conditions, "status", "conditions")).To(Succeed())
```

File: internal/controller/projectdevelopmentstream_controller_test.go (hunk 1 above)
Comment (ifireball) [reply internal/controller/projectdevelopmentstream_controller_test.go]: lets try to demonstrate as many of these cases as we can using the existing example-based system (without major code changes, but with additional files as needed), then we can review what we are left with and consider what it the best system for those tests.

I'm sorry, maybe its subjective but I find these pure code based tests hard to follow.
